lib/hydra/works/services/generic_file/persist_derivatives_output_file.rb

```diff
@@ -0,0 +1,26 @@
+require 'hydra/derivatives'
+
+module Hydra::Works
+  class PersistDerivativesOutputFile < Hydra::Derivatives::PersistOutputFileService
+
+    ##
+    # Persists a derivative to a GenericFile
+    # This Service conforms to the signature of `Hydra::Derivatives::PersistOutputFileService`.  
+    # The purpose of this Service is for use as an alternative to the default Hydra::Derivatives::PersistOutputFileService.  It's necessary because the default behavior in Hydra::Derivatives assumes that you're using LDP Basic Containment. Hydra::Works::GenericFiles use IndirectContainment.  This Service handles that case.
+    # 
+    # @param [Hydra::Works::GenericFile::Base] object the file will be added to
+    # @param [File] file the dervivative filestream
+    # @param [String] destination_name path to the file
+    # @option opts [RDF::URI or String] type URI for the RDF.type that identifies the file's role within the generic_file
+    # @option opts [Boolean] update_existing when set to false, always adds a new file.  When set to true, performs a create_or_update
+    # @option opts [Boolean] versioning whether to create new version entries
+
+    def self.call(object, file, destination_name, opts={})
+      type = opts[:type] || ::RDF::URI("http://pcdm.org/ServiceFile")
```

Comment:
I believe the URI here should be `http://pcdm.org/use#ServiceFile` per https://github.com/duraspace/pcdm/blob/master/pcdm-ext/use.rdf#L50-L57